plugin/pulpcore/plugin/download/asyncio/base.py

```diff
@@ -0,0 +1,153 @@
+from collections import namedtuple
+import hashlib
+import os
+import tempfile
+
+from pulpcore.app.models import Artifact
+from .exceptions import DigestValidationError, SizeValidationError
+
+
+DownloadResult = namedtuple('DownloadResult', ['url', 'artifact_attributes', 'path'])
+"""
+Args:
+    url (str): The url corresponding with the download.
+    path (str): The absolute path to the saved file
+    artifact_attributes (dict): Contains keys corresponding with pulpcore.plugin.Artifact fields.
+        This includes the computed digest values along with size information.
+"""
+
+class BaseDownloader(object):
```

Comment:
All objects are new-style in python3.  Don't need the `(object)`.

Reply:
Fixed in next push and throughout PR.